src/oop/special_class_method/hard/q3/MainTest.java

```diff
@@ -0,0 +1,249 @@
+        assertTrue(testClass.hasField(doubleFieldName, double.class),
+                "Your " + doubleFieldName + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(doubleFieldName, "private"),
+                "Your " + doubleFieldName + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(stringFieldName),
+                "Your " + className + " class is missing the " + stringFieldName + " attribute, or it is spelled incorrectly.");
+        assertTrue(testClass.hasField(stringFieldName, String.class),
+                "Your " + stringFieldName + " does not have the correct type.");
+        assertTrue(testClass.hasModifier(stringFieldName, "private"),
+                "Your " + stringFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(543.34, "Long"),
+                Arguments.of(222342.43, "Round"),
+                Arguments.of(3.3, "Tall"),
+                Arguments.of(4323.4, "Short")
+
+        );
+    }
+
+    @Test
+    public void chairClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {double.class, String.class};
+        assertTrue(testClass.hasConstructor(classArguments),
+                "Your " + className + " constructor does not have the correct parameters.");
+        assertTrue(testClass.hasModifier(classArguments, "public"),
+                "Your " + className + " constructor does not have the correct modifier.");
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void chairClassHasCorrectConstructor(double value1, String value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, String.class}
+        };
+        Class<?>[] constructorClasses = {double.class, String.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + stringFieldName + " attribute to the correct value.";
+        String incorrectConstructorVisibilityModifier =
+                "Your " + className + " constructor does not have the correct visibility modifier.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, stringFieldName),
+                incorrectStringValueMessage);
+        assertTrue(testClass.hasModifier(constructorClasses, "public"), incorrectConstructorVisibilityModifier);
```

Comment:
As you already tested this in the `chairClassHasRequiredConstructor` test, you do not need to test it again here.